catalog/opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchParserImpl.java

```diff
@@ -0,0 +1,716 @@
+            fail("URL is not valid: " + mue.getMessage());
+        }
+        LOGGER.info("URL after contextual population: {}", urlStr);
+    }
+
+    @Test
+    public void populateContextualEmptyMap() {
+        Map<String, String> searchPhraseMap = new HashMap<>();
+
+        String urlString = URL.toString();
+        assertThat(urlString, containsString(OpenSearchParserImpl.SEARCH_TERMS));
+        WebClient webClient = WebClient.create(URL.toString());
+        openSearchParserImpl.populateContextual(webClient,
+                searchPhraseMap,
+                Arrays.asList(
+                        "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort".split(
+                                ",")));
+        String urlStr = webClient.getCurrentURI()
+                .toString();
+        assertThat(urlStr, not(containsString("q=")));
+        assertThat(urlStr, containsString(OpenSearchParserImpl.SEARCH_TERMS));
+        try {
+            new URL(urlStr);
+        } catch (MalformedURLException mue) {
+            fail("URL is not valid: " + mue.getMessage());
+        }
+        LOGGER.info("URL after contextual population: {}", urlStr);
+    }
+
+    @Test
+    public void populateContextualNullMap() {
+        String urlString = URL.toString();
+        assertThat(urlString, containsString(OpenSearchParserImpl.SEARCH_TERMS));
+        WebClient webClient = WebClient.create(URL.toString());
+        openSearchParserImpl.populateContextual(webClient,
+                null,
+                Arrays.asList(
+                        "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort".split(
+                                ",")));
+        String urlStr = webClient.getCurrentURI()
+                .toString();
+        assertThat(urlStr, not(containsString("q=")));
+        assertThat(urlStr, containsString(OpenSearchParserImpl.SEARCH_TERMS));
+        try {
+            new URL(urlStr);
+        } catch (MalformedURLException mue) {
+            fail("URL is not valid: " + mue.getMessage());
+        }
+        LOGGER.info("URL after contextual population: {}", urlStr);
+    }
+
+    /**
+     * Verify that passing in null will still remove the parameters from the URL.
+     */
+    @Test
+    public void populateNullContextual() {
+        WebClient webClient = WebClient.create(URL.toString());
+        openSearchParserImpl.populateContextual(webClient,
+                null,
+                Arrays.asList(
+                        "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort".split(
+                                ",")));
+        String urlStr = webClient.getCurrentURI()
+                .toString();
+        assertThat(urlStr, not(containsString("?" + OpenSearchParserImpl.SEARCH_TERMS)));
+    }
+
+    @Test
+    public void populateTemporal() {
+        // TODO have actual set time strings to compare to
```

Comment:
I agree that it doesn't need to be fixed in this PR, but since we're here, would you please align it with our [code comment guidelines](https://codice.atlassian.net/wiki/display/DDF/Code+Comment+Guidelines) by creating a ticket if one doesn't exist and adding the ticket number to the comment?